### ISSUE TITLE
In grid view, enable filter by task instance state.

### ISSUE THREAD

**Issue by Pad71** [apache/airflow#38032]
### Description

For large dags it is difficult to find e.g. failed task instances in the grid. Filtering task instances by state would make this much easier.
The system would allow to 
- select the state to filter and then display only the task instances with the selected state
- remove filter and display all task instances

### Use case/motivation

Management and orientatiton in larger dags

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
You can hover over the states in the top filter bar, but yes, we only highlight tasks we don't yet filter them.

**Comment by Pad71** [comment]
> You can hover over the states in the top filter bar, but yes, we only highlight tasks we don't yet filter them.

Yeah, I know, but it's only applicable to dags with a small number of quests. 
We use airflow to load a large data warehouse and we usually have tens to hundreds of tasks in our dags.
And for these dags, it's not usable because the highlighting disappears when you have to scroll in the grid to see the other tasks.

It could be usable for example if the status icon could be clicked to set the highlight (instead of hover) and then clicked again to cancel highlightning .
